### DISCUSSION TITLE
How to import custom Python modules to the config.py?

### DISCUSSION THREAD

**Discussion by Marlpzg** [NVIDIA-NeMo/Guardrails#595] [Q&A] [answered]
Hi everyone, currently I was creating a Python module to implement the DataStore interface for a customized RedisStore usage.
The redis_store.py module is in the same location as the config.py inside of a configuration folder:

![image](https://github.com/NVIDIA/NeMo-Guardrails/assets/16272171/f8b45242-8114-467c-864e-dbe44cecc3c0)

The redis_store.py file only contains the DataStore implementation (which is currently a WIP):

```
from typing import Optional
from redis import asyncio as aioredis
from nemoguardrails.server.datastore.datastore import DataStore

class RedisStore(DataStore):
    def __init__(
        self, 
        url: str, 
        username: Optional[str] = None,
        password: Optional[str] = None
    ):
        self.url = url
        self.username = username
        self.password = password
        self.client = aioredis.from_url(
            url=url, username=username, password=password, decode_responses=True
        )

    async def set(self, key: str, value: str):
        await self.client.set(key, value)

    async def get(self, key: str) -> Optional[str]:
        return await self.client.get(key)
```

I imported it to the config.py like this:

```
# ... Some code

from redis_store import RedisStore

# More code...
```

But I end up getting **ModuleNotFoundError: No module named 'redis_store'**

Is there a specific way to deal with custom Python modules? Am I mising something?
I would like to find a way to solve this because my code in the config.py file is starting to accumulate and is getting hard to maintain.

Thanks in advance!

**Comment by PeerBoerner1** [comment, marked as answer]
This "module not found" is typically Python related and can happen for a variety of reasons, though it usually happens when your default directory is not what you think it is.  One way to solve this problem is to set the PYTHONPATH environment variable to point to your project directory where all of your "main" python logic is stored. This will force python to check that directory for any modules and packages in its associated subdirectories. You must also ensure that your python package subdirectories have __init__.py files present (marking the directory as a module). These __init__.py files can be empty. Hope this helps.
